### ISSUE TITLE
Error creating user without a group

### ISSUE THREAD

**Issue by bsemperlotti** [atlassian/terraform-provider-artifactory#4]
In the file `resource_artifactory_user.go`, the parameter `groups` is optional: 
```go
"groups": {
  Type:     schema.TypeSet,
  Elem:     &schema.Schema{Type: schema.TypeString},
  Set:      schema.HashString,
  Optional: true,
},
```
but when creating the resource with terraform, an error occurs :

```tf
resource artifactory_user "test" {
  name = "test"
  email = "test@example.com"
  admin = false
  profile_updatable = true
  disable_ui_access = false
  internal_password_disabled = false
}
```

```
artifactory_user.test: Creating...
  admin:                      "" => "false"
  disable_ui_access:          "" => "false"
  email:                      "" => "test@example.com"
  internal_password_disabled: "" => "false"
  name:                       "" => "test"
  profile_updatable:          "" => "true"
Releasing state lock. This may take a few moments...

Error: Error applying plan:

1 error(s) occurred:

* artifactory_user.test: 1 error(s) occurred:

* artifactory_user.test: unexpected EOF
```

However, when adding a group, the creation works as expected:
```tf
resource artifactory_user "test" {
  name = "test"
  email = "test@example.com"
  admin = false
  profile_updatable = true
  disable_ui_access = false
  internal_password_disabled = false
  groups = [ "${artifactory_group.default_group.id}" ]
}
```

```
artifactory_user.test: Creating...
  admin:                      "" => "false"
  disable_ui_access:          "" => "false"
  email:                      "" => "test@example.com"
  groups.#:                   "" => "1"
  groups.2438132315:          "" => "default"
  internal_password_disabled: "" => "false"
  name:                       "" => "test"
  profile_updatable:          "" => "true"
artifactory_user.test: Creation complete after 1s (ID: test)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

**Comment by dillon-giacoppo** [comment]
This was caused by the Artifactory API omitting the groups field instead of returning an empty list which is not the behaviour of other lists. This has been fixed in release v1.2.2

**Comment by bsemperlotti** [comment]
Thanks
